macros/new_datacard_tt_ot.py

```diff
@@ -142,38 +145,52 @@
-    fake_cfgs_MC_1 = build_cfgs(['fakesMC1'], datasets_MC_fakes, variable, l1_FakeFactorApplication_Region_genuinetauMC, bins)
-    fake_cfgs_MC_2 = build_cfgs(['fakesMC2'], datasets_MC_fakes, variable, l2_FakeFactorApplication_Region_genuinetauMC, bins)
     
-    fake_cfgs_1 = build_cfgs(
-        ['fakes'+dataset.name[-1]+'1' for dataset in cfg.datasets.data_datasets], 
-        cfg.datasets.data_datasets, variable, l1_FakeFactorApplication_Region, bins)
-    fake_cfgs_2 = build_cfgs(
-        ['fakes'+dataset.name[-1]+'2' for dataset in cfg.datasets.data_datasets], 
-        cfg.datasets.data_datasets, variable, l2_FakeFactorApplication_Region, bins)
+    fake_cfgs_DY_1 = build_cfgs(['fakesDY1'], cfg.datasets.DY_datasets, variable, l1_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_DY_2 = build_cfgs(['fakesDY2'], cfg.datasets.DY_datasets, variable, l2_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_TT_1 = build_cfgs(['fakesTT1'], cfg.datasets.TT_datasets, variable, l1_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_TT_2 = build_cfgs(['fakesTT2'], cfg.datasets.TT_datasets, variable, l2_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_WJ_1 = build_cfgs(['fakesWJ1'], cfg.datasets.WJ_datasets, variable, l1_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_WJ_2 = build_cfgs(['fakesWJ2'], cfg.datasets.WJ_datasets, variable, l2_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_Diboson_1 = build_cfgs(['fakesDiboson1'], cfg.datasets.Diboson_datasets, variable, l1_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_Diboson_2 = build_cfgs(['fakesDiboson2'], cfg.datasets.Diboson_datasets, variable, l2_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_singleTop_1 = build_cfgs(['fakessingleTop1'], cfg.datasets.singleTop_datasets, variable, l1_FakeFactorApplication_Region_genuinetauMC, bins)
+    fake_cfgs_singleTop_2 = build_cfgs(['fakessingleTop2'], cfg.datasets.singleTop_datasets, variable, l2_FakeFactorApplication_Region_genuinetauMC, bins)
     
     fake_cfgs_Embedded_1 = build_cfgs(
         ['fakesEmbedded'+dataset.name[-1]+'1' for dataset in cfg.datasets.Embedded_datasets], 
         cfg.datasets.Embedded_datasets, variable, l1_FakeFactorApplication_Region_genuinetauMC_Embedded, bins)
     fake_cfgs_Embedded_2 = build_cfgs(
         ['fakesEmbedded'+dataset.name[-1]+'2' for dataset in cfg.datasets.Embedded_datasets], 
         cfg.datasets.Embedded_datasets, variable, l2_FakeFactorApplication_Region_genuinetauMC_Embedded, bins)
-    
+
+    for cfg in [fake_cfgs_DY_1,fake_cfgs_DY_2,fake_cfgs_TT_1,fake_cfgs_TT_2,fake_cfgs_WJ_1,fake_cfgs_WJ_2,fake_cfgs_Diboson_1,fake_cfgs_Diboson_2,fake_cfgs_singleTop_1,fake_cfgs_singleTop_2,fake_cfgs_Embedded_1,fake_cfgs_Embedded_2]:
+        cfg['scale'] = -1.
```

Comment:
Watch out, here you redefine `cfg` in the for loop, so that later `cfg` points to a `Component_cfg` object and not the `cfg` "config" module imported at the first place.